cmd/ecs-deploy/task_def.go

```diff
@@ -114,33 +116,104 @@ const (
 
 // ECRImage represents an ECR Image tag broken into its constituent parts
 type ECRImage struct {
-	AWSRegion      string
-	imageURI       string
-	ImageTag       string
-	RegistryID     string
-	RepositoryURI  string
-	RepositoryName string
+	AWSRegion        string
+	Digest           string
+	ImageURIByDigest string
+	ImageURIByTag    string
+	RegistryID       string
+	RepositoryURI    string
+	RepositoryName   string
+	Tag              string
+	IsTagBased       bool
+	IsDigestBased    bool
```

Comment:
I'm not a huge fan of having booleans. The truth here lies in the `ImageURIByDigest` or `ImageURIByTag`. Drop the booleans and modify the URI fields to be pointers. Here's what I'd like to see:

```suggestion
	AWSRegion        string
	Digest           string
	ImageURIByDigest *string
	ImageURIByTag    *string
	RegistryID       string
	RepositoryURI    string
	RepositoryName   string
	Tag              string
```

Reply:
👍